src/pybind/cephfs/cephfs.pyx

```diff
@@ -2165,6 +2165,7 @@ cdef class LibCephFS(object):
             buf = <char *>realloc_chk(buf, _size)
             with nogil:
                 ret = ceph_readlink(self.cluster, _path, buf, _size)
+                buf[ret] = 0
             if ret < 0:
                 raise make_ex(ret, "error in readlink")
             return buf
```

Comment:
Better I think would be:

```suggestion
            py_bytes = buf[:ret]
            return py_bytes
```

See also: "Besides not working for null bytes..." from https://cython.readthedocs.io/en/latest/src/tutorial/strings.html

Reply:
Thanks. I updated the commit like this.